modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java

```diff
@@ -179,19 +170,18 @@ public Response snapshot(@FormParam("mediapackage") final MediaPackageImpl media
               responseCode = SC_INTERNAL_SERVER_ERROR)},
       returnDescription = "No content is returned.")
   public Response delete(@PathParam("id") final String mediaPackageId) {
-    return handleException(new P1Lazy<Response>() {
-      @Override public Response get1() {
-        if (mediaPackageId != null) {
-          final AQueryBuilder q = getAssetManager().createQuery();
-          if (q.delete(AssetManager.DEFAULT_OWNER, q.snapshot()).where(q.mediaPackageId(mediaPackageId)).run() > 0) {
-            return noContent();
-          } else {
-            return notFound();
-          }
-        } else
-          return notFound();
+    if (StringUtils.isEmpty(mediaPackageId)) {
+      return notFound();
```

Comment:
Ugh, shouldn't this be badRequest()?  Nevermind, it's the same (incorrect) behaviour.